src/main/java/campus/infrastructure/Parser.java

```diff
@@ -0,0 +1,219 @@
+     */
+    public String parseMessage(String userInput) {
+        try {
+            String message = this.processCommand(userInput);
+            assert (message != null);
+            return message;
+        } catch (CampusException e) {
+            return this.ui.displayErrorMessage(e);
+        }
+    }
+
+    /**
+     * Process the command fed by the user and decides which path of execution to go next, also handles null cases
+     * and unknown command cases
+     * @param userInput the user command input
+     * @return boolean true for continue feeding, false only in the case of "bye" that is to exit the ChatBot
+     * @throws CampusException Exception in the case that the command is not understood
+     */
+    public String processCommand(String userInput) throws CampusException {
+        String msg = null;
+        String[] arr = userInput.trim().split(" ", 2);
+        String firstWord = arr[0].trim();
+        String remaining = arr.length > 1 ? arr[1].trim() : "";
+
+        switch (firstWord) {
+        case "list":
+            // fallthrough
+        case "mark":
+            // fallthrough
+        case "unmark":
+            // fallthrough
+        case "delete":
+            msg = handleUpdateCommands(firstWord, userInput);
+            break;
+        case "todo":
+            // fallthrough
+        case "deadline":
+            // fallthrough
+        case "event":
+            msg = handleTaskCommand(firstWord, remaining);
+            break;
+        case "bye":
+            return ui.printExiting();
+        case "":
+            break;
+        case "find":
+            msg = handleFindCommand(remaining);
+            break;
+        default:
+            throw new CampusException("Sorry, I don't understand that command,"
+                    + " please check for potential spelling errors");
+        }
+        this.storage.updateFileFromList(this.taskList);
+        assert (msg != null);
+        return msg;
+    }
+
+    private String handleTaskCommand(String command, String details) throws CampusException {
+        switch (command) {
+        case "todo":
+            return handleTodoCommand(details);
+        case "deadline":
+            return handleDeadlineCommand(details);
+        case "event":
+            return handleEventCommand(details);
+        default:
+            throw new CampusException("Invalid task command");
+        }
+    }
+
+    /**
+     * Function to handle the command of type 'find'
+     * @param remaining the remaining portion of the command line entered
+     */
+    public String handleFindCommand(String remaining) {
+        TaskList tempTaskList = this.taskList.getTaskListWhere(remaining);
+        return this.ui.printTaskList(tempTaskList);
+    }
+
+    /**
+     * Handles the commands for mark, unmark and delete - passes it to the taskList object which contains methods
+     * to deal with manipulation of the task
+     * @param command mark/unmark/delete
```

Comment:
The Java coding standard requires that punctuation should be present. This mistake can also be found in several other parts of the code.